src/ring_theory/adjoin_root.lean

```diff
@@ -154,4 +190,52 @@ lemma mul_div_root_cancel :
     f.map (of f) :=
 mul_div_eq_iff_is_root.2 $ is_root_root _
 
+section findim
+open vector_space
+open finite_dimensional
+
+/-- The restriction of `adjoin_root.mk f` to the polynomials of degree less than `f`,
+viewed as a linear map between vector spaces over `K`. -/
+def degree_lt_linear_map : degree_lt K (f.nat_degree) →ₗ[K] adjoin_root f :=
```

Comment:
I wonder whether this can be refactored in a way similar to what happened in the dependency PR.

Reply:
I bundled it all into on definition, although the proofs got slightly longer.